data/src/main/java/org/apache/iceberg/data/GenericAppenderFactory.java

```diff
@@ -42,10 +47,25 @@
 public class GenericAppenderFactory implements FileAppenderFactory<Record> {
 
   private final Schema schema;
+  private final PartitionSpec spec;
+  private final int[] equalityFieldIds;
+  private final Schema eqDeleteRowSchema;
+  private final Schema posDeleteRowSchema;
   private final Map<String, String> config = Maps.newHashMap();
 
-  public GenericAppenderFactory(Schema schema) {
+  public GenericAppenderFactory(Schema schema, PartitionSpec spec) {
+    this(schema, spec, null, schema, null);
```

Comment:
Why not set `eqDeleteRowSchema` to null since `equalityFieldIds` is null?

Reply:
OK, it's more reasonable to set the `eqDeleteRowSchema` to be null when `equalityFieldIds` is null.